upup/pkg/fi/cloudup/new_cluster.go

```diff
@@ -674,6 +674,16 @@ func setupMasters(opt *NewClusterOptions, cluster *api.Cluster, zoneToSubnetMap
 				g.Spec.Zones = []string{zone}
 			}
 
+			if api.CloudProviderID(cluster.Spec.CloudProvider) == api.CloudProviderAWS && (g.Spec.InstanceMetadata == nil || g.Spec.InstanceMetadata.HTTPTokens == nil) {
+				k8sVersion, err := version.ParseKubernetesVersion(cluster.Spec.KubernetesVersion)
+				if err == nil && version.IsKubernetesGTE("1.20", *k8sVersion) {
```

Comment:
Looks like it will be enabled on new clusters with k8s 1.20+. Do we really need the k8s check here when it is only for new clusters?

Reply:
I discussed a bit with @rifelpet and he is worried that this may break some existing clusters if operators don't read the release notes.
I would question a bit the requirement for 1.20+ and change to skip for debian 9 and flatcar images.

Reply:
If the intention is to only do this for new clusters, it won't break any existing ones (I hope)

Reply:
- [ ] I think historically we've made node-level cloud provider changes by only affecting new clusters and the newly supported k8s version. That way someone creating k8s 1.18 clusters wont have a change in behavior when upgrading from kops 1.18 to 1.19.

This might be especially important for changes that could break workloads like this one.

I don't nt have a particularly strong opinion about this though as long as the release notes make it abundantly clear. Given that this is for 1.20 perhaps we get more input at the mext office hours?

Reply:
Good point, Friday it is then. 😄

Reply:
I have observed that there are some components which break if IMDSv2 is enforced. eg: Kube2iam which does not support IMDSv2 properly. An upgrade could break workloads which depend on kube2iam for IAM access. Enforcing IMDSv2 on a cluster without the operator clearly knowing about it can make for a pretty bad day at the office....